go/kbfs/libkbfs/favorites.go

```diff
@@ -273,26 +300,47 @@ func (f *Favorites) closeReq(req *favReq, err error) {
 	req.err = err
 	close(req.done)
 	for _, fav := range req.toAdd {
-		delete(f.inFlightAdds, fav)
+		delete(f.inFlightAdds, fav.Favorite)
 	}
 }
 
 // sendChangesToEditHistory notes any deleted favorites and removes them
 // from this user's kbfsedits.UserHistory.
-func (f *Favorites) sendChangesToEditHistory(oldCache map[Favorite]bool) {
+func (f *Favorites) sendChangesToEditHistory(oldCache map[Favorite]favoriteData) {
 	for oldFav := range oldCache {
-		if !f.cache[oldFav] {
+		if _, present := f.favCache[oldFav]; !present {
 			f.config.UserHistory().ClearTLF(tlf.CanonicalName(oldFav.Name),
 				oldFav.Type)
 		}
 	}
-	for newFav := range f.cache {
-		if !oldCache[newFav] {
+	for newFav := range f.favCache {
+		if _, present := oldCache[newFav]; !present {
 			f.config.KBFSOps().RefreshEditHistory(newFav)
 		}
 	}
 }
 
+type favoriteType int
+
+const (
+	typeIgnored favoriteType = iota
+	typeNew
+	typeFavorite
+)
+
+func toFolder(fav Favorite, data favoriteData,
+	favType favoriteType) keybase1.Folder {
+	return keybase1.Folder{
+		Name:    fav.Name,
+		Private: data.Private,
+		// TODO: figure out what the deal is here
+		NotificationsOn: favType == typeFavorite,
```

Comment:
Oof, `NotificationsOn` is super old.  I'm pretty sure it's currently unused in the code.  You can probably just remove it from the protocol.  I don't even remember what it was for.